xbmc/addons/AddonDll.cpp

```diff
@@ -461,5 +400,31 @@ ADDON_STATUS CAddonDll::TransferSettings()
   return ADDON_STATUS_OK;
 }
 
+bool CAddonDll::CheckAPIVersion(int type)
+{
+  /* check the API version */
+  const char* kodiVersion = kodi::addon::GetTypeVersion(type);
+  const char* addonVersion = m_pDll->GetAddonTypeVersion(type);
+
+  if (AddonVersion(addonVersion) != AddonVersion(kodiVersion))
+  {
+    CLog::Log(LOGERROR, "Add-on '%s' is using an incompatible API version for type '%s'. Kodi API version = '%s', add-on API version '%s'",
+                            Name().c_str(),
+                            kodi::addon::GetTypeName(type),
+                            kodiVersion,
+                            addonVersion);
+
+    std::string text = StringUtils::Format(g_localizeStrings.Get(29803).c_str(),
+                                           kodi::addon::GetTypeName(type),
+                                           kodiVersion,
+                                           addonVersion);
+    CEventLog::GetInstance().AddWithNotification(EventPtr(new CNotificationEvent(Name(), text, EventLevel::Error)));
```

Comment:
@tamland have changed it to the `CEventLog::GetInstance().AddWithNotification` with:
![bildschirmfoto vom 2017-04-27 02-04-08](https://cloud.githubusercontent.com/assets/6879739/25462752/68eb6e6a-2af1-11e7-97c0-989079a57a5e.png)

This by start of screensaver with wrong version:
![bildschirmfoto vom 2017-04-27 02-24-33](https://cloud.githubusercontent.com/assets/6879739/25462756/72b6aea0-2af1-11e7-91bb-f45eac56a1ae.png)